pkglistgen/tool.py

```diff
@@ -703,8 +703,13 @@ def update_and_solve_target(
             else:
                 if os.path.exists(cache_dir):
                     shutil.rmtree(cache_dir)
-                logging.debug(subprocess.check_output(
-                    ['git', 'clone', '--recurse-submodules', git_url, cache_dir], encoding='utf-8'))
+                split = git_url.split('#')
+                git_url = split[0]
+                args = ['git', 'clone', '--recurse-submodules', git_url, cache_dir]
+                if len(split) == 2:
+                    args.append('--branch')
```

Comment:
`--single-branch` as well (unconditionally)?

Reply:
we can do so ... added another commit for that as it is independend of the branch selection.